storage/storage-interface/src/lib.rs

```diff
@@ -558,6 +566,87 @@ impl<PS: ProtocolSpec> MoveStorage for &dyn DbReader<PS> {
             .collect()
     }
 
+    fn fetch_config_by_version(&self, config_id: ConfigID, version: Version) -> Result<Vec<u8>> {
```

Comment:
looks like you duplicated the following logic at least 2 more places, something not right.

Reply:
You are right. I'll find a way to move trait `MoveStorage` to storage-interface so that I can implement `MoveStorage` for all types implementing `DbReader`, instead of duplicating code.